runtime/lua/vim/diagnostic.lua

```diff
@@ -429,32 +429,30 @@ local function get_diagnostics(bufnr, opts, clamp)
     end
   end
 
+  local add_iterator = function(buf_nr, diag_tbl)
```

Comment:
- prefer `local function foo()` except when there's a technical reason the `foo  = function` form is needed. I thought this was checked by lint...
- "iterator" (or more preferred: "iter") in a function name implies it returns a Lua "iterator". this function doesn't do that, so the name is confusing. 
    - for ad-hoc locally-scoped functions like this, smaller names are often _less_ confusing than verbose names. Avoid "premature explanation".